diff --git a/drivers/base/firmware_loader/main.c b/drivers/base/firmware_loader/main.c
index bf44c79be..c39141faa 100644
--- a/drivers/base/firmware_loader/main.c
+++ b/drivers/base/firmware_loader/main.c
@@ -780,7 +780,8 @@ _request_firmware(const struct firmware **firmware_p, const char *name,
                        dev_warn(device,
                                 "Direct firmware load for %s failed with error %d\n",
                                 name, ret);
-               ret = firmware_fallback_sysfs(fw, name, device, opt_flags, ret);
+               if (opt_flags & FW_OPT_USERHELPER)
+                       ret = firmware_fallback_sysfs(fw, name, device, opt_flags, ret);
        } else
                ret = assign_fw(fw, device, opt_flags);
 
