
Index: misdk/linuxkernel/4.19/drivers/pci/controller/pcie-mstar.c
===================================================================
--- misdk.orig/linuxkernel/4.19/drivers/pci/controller/pcie-mstar.c
+++ misdk/linuxkernel/4.19/drivers/pci/controller/pcie-mstar.c
@@ -230,6 +230,9 @@
 #define REC_RXCFG_TXTS2_NUM_MASK		(0x7FUL << 16)
 #define REC_RXCFG_TXTS2_NUM				(0x40UL << 16)
 
+#define REG_MMIO_LOCK					0x564
+#define REG_MMIO_WR_LOCK_EN				BIT(0)
+#define REG_MMIO_RD_LOCK_EN				BIT(1)
 
 #define AHB2PCIE_BASEL(base)		    (base & 0xffffffff)
 #define AHB2PCIE_BASEH(base)		    (base >> 32)
@@ -1245,6 +1248,10 @@
 	pcie_write(port, pcie_read(port, PCI_RSTCR) |
 		PCI_RST_DEASSERTED, PCI_RSTCR);
 
+	/* Configure MEM write lock, mem read will wait mem write done */
+	pcie_write(port, pcie_read(port, REG_MMIO_LOCK) |
+		REG_MMIO_WR_LOCK_EN, REG_MMIO_LOCK);
+
 	/* init miu0 address */
 	tVal = MIU0_BUS_BASE_ADDR >> 20;
 	pcie_riu_writew(port, tVal, REG_MIU0_START_ADDR0);
