diff --git a/meta_build/load_meta/host/wlan_host/qcom_cfg.ini b/meta_build/load_meta/host/wlan_host/qcom_cfg.ini
index a80baba..51002ec 100755
--- a/meta_build/load_meta/host/wlan_host/qcom_cfg.ini
+++ b/meta_build/load_meta/host/wlan_host/qcom_cfg.ini
@@ -72,7 +72,10 @@ gIPADescSize=800
 dp_tx_ring_size=3072
 gReorderOffloadSupported=1
 gCEClassifyEnable=1
-rx_mode=20
+rx_mode=16
+dp_poll_mode_enable=1
+dp_napi_enabled=0
+gInterfaceChangeWait=0
 gEnableFastPath=1
 gEnableIpTcpUdpChecksumOffload=1
 TSOEnable=1
diff --git a/chss_host_LEA/chss_proc/host/AIO/drivers/qca-wifi-host-cmn/hif/src/ce/ce_assignment.h b/chss_host_LEA/chss_proc/host/AIO/drivers/qca-wifi-host-cmn/hif/src/ce/ce_assignment.h
index 5191364..26d6d9c 100644
--- a/chss_host_LEA/chss_proc/host/AIO/drivers/qca-wifi-host-cmn/hif/src/ce/ce_assignment.h
+++ b/chss_host_LEA/chss_proc/host/AIO/drivers/qca-wifi-host-cmn/hif/src/ce/ce_assignment.h
@@ -1265,13 +1265,13 @@ static struct CE_pipe_config target_ce_config_wlan_qca6390[] = {
 #define QCA_6490_CE_COUNT 9
 static struct CE_attr host_ce_config_wlan_qca6490[] = {
 	/* host->target HTC control and raw streams */
-	{ /* CE0 */ CE_ATTR_FLAGS, 0, 16, 2048, 0, NULL,},
+	{ /* CE0 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 16, 2048, 0, NULL,},
 	/* target->host HTT + HTC control */
-	{ /* CE1 */ CE_ATTR_FLAGS, 0, 0,  2048, 512, NULL,},
+	{ /* CE1 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 0,  2048, 512, NULL,},
 	/* target->host WMI */
-	{ /* CE2 */ CE_ATTR_FLAGS, 0, 0,  3200, 64, NULL,},
+	{ /* CE2 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 0,  3200, 64, NULL,},
 	/* host->target WMI */
-	{ /* CE3 */ CE_ATTR_FLAGS, 0, 32, 2048, 0, NULL,},
+	{ /* CE3 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 32, 2048, 0, NULL,},
 	/* host->target HTT */
 	{ /* CE4 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR), 0,
 		CE_HTT_H2T_MSG_SRC_NENTRIES_QCA6490, 256, 0, NULL,},
@@ -1279,15 +1279,15 @@ static struct CE_attr host_ce_config_wlan_qca6490[] = {
 #ifdef REMOVE_PKT_LOG
 	{ /* CE5 */ 0, 0, 0, 0, 0, NULL,},
 #else
-	{ /* CE5 */ CE_ATTR_FLAGS, 0, 0, 2048, 512, NULL,},
+	{ /* CE5 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 0, 2048, 512, NULL,},
 #endif
 	/* Target autonomous HIF_memcpy */
-	{ /* CE6 */ CE_ATTR_FLAGS, 0, 0, 0, 0, NULL,},
+	{ /* CE6 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 0, 0, 0, NULL,},
 	/* ce_diag, the Diagnostic Window */
 	{ /* CE7 */ (CE_ATTR_DIAG_FLAGS | CE_ATTR_DISABLE_INTR), 0,
 		0, DIAG_TRANSFER_LIMIT, 0, NULL,},
 	/* Reserved for target */
-	{ /* CE8 */ CE_ATTR_FLAGS, 0, 0, 0, 0, NULL,},
+	{ /* CE8 */ (CE_ATTR_FLAGS | CE_ATTR_DISABLE_INTR | CE_ATTR_ENABLE_POLL), 0, 0, 0, 0, NULL,},
 	/* CE 9, 10, 11 belong to CoreBsp & MHI driver */
 };
 
